mesonbuild/interpreter.py

```diff
@@ -2307,7 +2312,10 @@ def func_configure_file(self, node, args, kwargs):
             # We use absolute paths for input and output here because the cwd
             # that the command is run from is 'unspecified', so it could change.
             # Currently it's builddir/subdir for in_builddir else srcdir/subdir.
-            values = mesonlib.get_filenames_templates_dict([ifile_abs], [ofile_abs])
+            if ifile_abs:
+                values = mesonlib.get_filenames_templates_dict([ifile_abs], [ofile_abs])
+            else:
+                values = mesonlib.get_filenames_templates_dict(None, [ofile_abs])
```

Comment:
Maybe, but I think this should be a separate commit/PR then. The current code clearly allows only a single input file intentionally.